controller/src/test/java/io/pravega/controller/rest/v1/PingTest.java

```diff
@@ -63,8 +69,57 @@ public void tearDown() {
 
     @Test
     public void test() {
-        String streamResourceURI = "http://localhost:" + serverConfig.getPort() + "/ping";
+        String streamResourceURI = getURLScheme() + "://localhost:" + serverConfig.getPort() + "/ping";
         Response response = client.target(streamResourceURI).request().buildGet().invoke();
         assertEquals(200, response.getStatus());
     }
+
+    protected abstract String getURLScheme();
+
+    public static class SimplePingTest extends PingTest {
+
+        @Override
+        protected Client createJerseyClient() {
+            return ClientBuilder.newClient();
+        }
+
+        @Override
+        RESTServerConfig getServerConfig() {
+            return RESTServerConfigImpl.builder().host("localhost").port(TestUtils.getAvailableListenPort())
+                                .build();
+        }
+
+        @Override
+        protected String getURLScheme() {
+            return "http";
+        }
+    }
+
+    public static class SecurePingTest extends PingTest {
+
+        @Override
+        protected Client createJerseyClient() {
+            SslConfigurator sslConfig = SslConfigurator.newInstance()
+                                                       .trustStoreFile("../config/bookie.truststore.jks");
```

Comment:
It just seems bad to hardcode these paths into a test because it presupposed the working directory.   I find that Gradle and IntelliJ have different defaults in that respect.  I think the ideal would be for these files to be in `src/test/resources` and extracted as temp files.

Reply:
We need these files for easy running of `standalone`. Reusing these here.

Reply:
I feel that @arvindkandhare has not addressed the comment. The key to @EronWright 's comment is:

>  it presupposed the working directory. I find that Gradle and IntelliJ have different defaults in that respect.

My recommendation is that you create a temporary file for this test, populate it, use it in the test, and drop it.